service/src/test/java/greencity/mapping/MultipartBase64ImageMapperTest.java

```diff
@@ -0,0 +1,106 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Base64;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(MockitoExtension.class)
+class MultipartBase64ImageMapperTest {
+
+    @InjectMocks
+    private MultipartBase64ImageMapper mapper;
+    private String validBase64Image;
+    private File tempFile;
+
+    @BeforeEach
+    void setUp() {
+        validBase64Image = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg==";
+        tempFile = new File("tempImage.jpg");
+    }
+
+    @AfterEach
+    void tearDown() {
+        if (tempFile != null && tempFile.exists() && !tempFile.delete()) {
+            System.err.println("Failed to delete temporary file.");
+        }
+    }
+
+    @Test
+    void convert_ValidBase64Image_ShouldReturnMultipartFile() {
+        String base64Image = validBase64Image;
+
+        MultipartFile result = mapper.convert(base64Image);
+
+        assertNotNull(result);
+        assertInstanceOf(MultipartFileImpl.class, result);
+        assertEquals("mainFile", result.getName());
+        assertEquals("tempImage.jpg", result.getOriginalFilename());
+        assertTrue(result.getSize() > 0);
+
+        try {
+            assertNotNull(result.getBytes());
+            assertTrue(result.getBytes().length > 0);
+        } catch (IOException e) {
+            fail("Failed to get bytes from MultipartFile: " + e.getMessage());
+        }
+    }
+
+    @Test
+    void convert_NullInput_ShouldThrowException() {
+        String base64Image = null;
+
+        assertThrows(Exception.class, () -> mapper.convert(base64Image));
+    }
+
+    @Test
+    void convert_EmptyString_ShouldThrowException() {
+        String base64Image = "";
+
+        assertThrows(Exception.class, () -> mapper.convert(base64Image));
+    }
+
+    @Test
+    void convert_Base64WithoutDataPrefix_ShouldThrowException() {
+        String base64Image = Base64.getEncoder().encodeToString("test".getBytes());
+
+        assertThrows(Exception.class, () -> mapper.convert(base64Image));
+    }
+
+    @Test
+    void convert_ValidJpegBase64Image_ShouldReturnMultipartFile() {
+        String jpegBase64 = "data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEAYABgAAD/2wBDAAEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQH/2wBDAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQH/wAARCAABAAEDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAv/xAAUEAEAAAAAAAAAAAAAAAAAAAAA/8QAFQEBAQAAAAAAAAAAAAAAAAAAAAX/xAAUEQEAAAAAAAAAAAAAAAAAAAAA/9oADAMBAAIRAxEAPwA/8A8A";
+
+        MultipartFile result = mapper.convert(jpegBase64);
+
+        assertNotNull(result);
+        assertEquals("mainFile", result.getName());
+        assertEquals("tempImage.jpg", result.getOriginalFilename());
+        assertTrue(result.getSize() > 0);
+    }
+
+
+    @Test
+    void convert_ValidImage_ShouldCleanupTempFile() {
```

Comment:
it doesn't really test if the mapper cleans the file up, though?

```suggestion
    void convert_ValidImage_ShouldCreateTempFile() {
```

(this is the same suggestion that rabbit did, lol.)